src/main/java/moze_intel/projecte/emc/SimpleStack.java

```diff
@@ -76,6 +82,9 @@ public boolean equals(Object obj)
 			}
 
 			//return this.id == other.id && this.damage == other.damage;
+			if(this.nbt != null && other.nbt != null){
+				return this.id == other.id && this.qnty == other.qnty && this.damage == other.damage && this.nbt.equals(other.nbt);
+			}
 			return this.id == other.id && this.qnty == other.qnty && this.damage == other.damage;
```

Comment:
This return should only happen, when `this.nbt == null && other.nbt == null`, i think. (The Stacks are not the same, when one has NBT and the other has not!)